File: Assets/AltUnityTester/AltUnityDriver/Commands/AltUnityCommands/AltUnitySetNotification.cs
using Altom.AltUnityDriver.Notifications;

namespace Altom.AltUnityDriver.Commands
{
    public class AltUnitySetNotification : AltBaseCommand
    {
        private readonly AltUnitySetNotificationParams cmdParams;

        public AltUnitySetNotification(IDriverCommunication commHandler, NotificationType notificationType) : base(commHandler)
        {
            this.cmdParams = new AltUnitySetNotificationParams(notificationType);
        }
        public void Execute()
        {
            this.CommHandler.Send(this.cmdParams);
            var data = this.CommHandler.Recvall<string>(this.cmdParams);
            ValidateResponse("Ok", data);
        }
    }
}

File: Bindings~/java/src/main/java/ro/altom/altunitytester/Commands/AltUnityCommands/AltUnitySetNotification.java
package ro.altom.altunitytester.Commands.AltUnityCommands;

import ro.altom.altunitytester.IMessageHandler;
import ro.altom.altunitytester.Commands.AltBaseCommand;

public class AltUnitySetNotification extends AltBaseCommand {
    private AltUnitySetNotificationParams cmdParams;

    public AltUnitySetNotification(IMessageHandler messageHandler,
            AltUnitySetNotificationParams altUnitySetNotificationParams) {
        super(messageHandler);
        this.cmdParams = altUnitySetNotificationParams;
    }

    public void Execute() {
        cmdParams.setCommandName("setNotification");
        SendCommand(cmdParams);
        recvall(cmdParams, String.class);
    }
}
